### PR TITLE
Updating .NET samples to newer SDK versions

### PR DESCRIPTION
**Pull request by bitsforthought**
Updated the .NET How-to to version 3. Also updated the .NET Getting Started app to version 2.0-preview, but this is just a stepping stone -- I'll update it to version 3 in a subsequent PR.

FYI @liamca @ashmaka @chaosrealm @Yahnoosh @mhko

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,10 +32,10 @@ The second sample AzureSearchDotNetSample is a more detailed example that demons
 
 ## More information
 
-For more details on these samples, please refer to these articles on Azure.com:
+For more details on these samples, please refer to these articles:
 
-  - [How to use Azure Search from a .NET Application](https://azure.microsoft.com/en-us/documentation/articles/search-howto-dotnet-sdk/).
-  - [Get started with your first Azure Search application in .NET](https://azure.microsoft.com/en-us/documentation/articles/search-get-started-dotnet/).
+  - [How to use Azure Search from a .NET Application](https://docs.microsoft.com/azure/search/search-howto-dotnet-sdk).
+  - [Get started with your first Azure Search application in .NET](https://docs.microsoft.com/azure/search/search-howto-dotnet-sdk).
```
Comment (mhko) [inline README.md]: The two links point to the same page. Were the old links pointing to the same page before?

**Review by mhko — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (bitsforthought) [reply README.md]: Whoops, I'll fix that

Comment (bitsforthought) [reply README.md]: It turns out the second ACOM article was deleted, so both links re-directed to the same article. I've just removed the second link.

**Review by liamca — COMMENTED** [review]
LGTM

**Pull request merged**